packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -44,21 +45,49 @@ export function set_inspect_effects(v) {
 	inspect_effects = v;
 }
 
+/** @type {null | Set<() => void>} */
+let onchange_batch = null;
+
+/**
+ * @param {Function} fn
+ */
+export function batch_onchange(fn) {
+	// @ts-expect-error
+	return function (...args) {
+		let previous_onchange_batch = onchange_batch;
+
+		try {
+			onchange_batch = new Set();
+
+			// @ts-expect-error
+			return fn.apply(this, args);
+		} finally {
+			for (const onchange of /** @type {Set<() => void>} */ (onchange_batch)) {
+				onchange();
+			}
+
+			onchange_batch = previous_onchange_batch;
+		}
+	};
+}
+
 /**
  * @template V
  * @param {V} v
+ * @param {ValueOptions} [o]
  * @param {Error | null} [stack]
  * @returns {Source<V>}
  */
-export function source(v, stack) {
+export function source(v, o, stack) {
 	/** @type {Value} */
 	var signal = {
 		f: 0, // TODO ideally we could skip this altogether, but it causes type errors
 		v,
 		reactions: null,
 		equals,
 		rv: 0,
-		wv: 0
+		wv: 0,
+		o
```

Comment:
Did you run the local benchmark to see how costly this is?

Reply:
Didn't benchmarks run on CI? Let me check locally.

Reply:
So apparently it did add a bit of footprint

**ONCHANGE BRANCH**
```
Finished reactivity benchmarks.

{ suite_time: '5243.41', suite_gc_time: '119.39' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5249.75', suite_gc_time: '120.65' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5272.65', suite_gc_time: '119.19' }
```

**MAIN BRANCH**
```
Finished reactivity benchmarks.

{ suite_time: '5483.77', suite_gc_time: '55.10' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5550.95', suite_gc_time: '53.56' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5490.34', suite_gc_time: '188.93' }
```

not sure how to fix this

Reply:
Mmm i've run them again after merging main into onchange and somehow main got worse? :D

Either i'm doing something wrong or it's just random variation :)

**MAIN**
```
Finished reactivity benchmarks.

{ suite_time: '5461.84', suite_gc_time: '164.92' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5543.81', suite_gc_time: '53.33' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5434.36', suite_gc_time: '147.72' }
```

**ONCHANGE**
```
Finished reactivity benchmarks.

{ suite_time: '5404.90', suite_gc_time: '119.74' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5313.96', suite_gc_time: '119.41' }
```
```
Finished reactivity benchmarks.

{ suite_time: '5311.29', suite_gc_time: '118.71' }
```